### PR TITLE
OSX support added

### PR DESCRIPTION
**Pull request by r0ssIV**
Hi,

Your solution seems a great alternative for vagrant rsync-auto.
Revised the script to support OSX+fswatch for monitoring file changes.
Will gladly explain all the changes made.

### CODE REVIEW

**Comment by replete** [comment]
This looks great, did it work okay?

File: watch.sh [hunk 1]
```diff
@@ -12,25 +12,17 @@
 #
 # ./watch.sh rsync -Cra --out-format='[%t]--%n' --include core \
 #  --WATCH_EXCLUDE=sites/default/files --delete ../web/ vagrant@d8.local:/var/www
+#------------------------------------------------------------
+# edited by Ross Ivantsiv to support OSX+fswatch (Darwin)
+# (c) DataSyntax PE, ross [at] datasyntax.ua
```
Comment (fago) [inline watch.sh]: Let's just add it at the top so all (c) are next to each other?

File: README.md [hunk 2]
```diff
@@ -1,16 +1,41 @@
...
+fsevents
+```
+brew tap homebrew/dupes
+brew install rsync
+brew install fswatch
+```
+
+Windows
+_______
+No bash version of the project exist for Windows. Instead we are developing a C# clone.
```
Comment (fago) [inline README.md]: cool - where can this be found?

File: README.md [hunk 3]
```diff
@@ -18,7 +43,7 @@ Installation
  * Clone the script in a directory of your choice, e.g.
 ```
 cd ~/opt
-git clone git@github.com:drunomics/syncd.git
+git clone git@github.com:DataSyntax/syncd.git
```
Comment (fago) [inline README.md]: shouldn't be changed in the PR as the repo stays then

File: README.md [hunk 4]
```diff
@@ -1,16 +1,41 @@
+
+Project was originally forked from https://github.com/drunomics/syncd which was designed for Linux systems.
```
Comment (fago) [inline README.md]: should be removed for a merge :)

File: syncd [hunk 5]
```diff
@@ -17,15 +19,32 @@ while [ ! -e $CONF_FILE ]; do
...
 
 . $CONF_FILE
 
-COMMAND="rsync $RSYNC_OPTIONS --exclude=$RSYNC_EXCLUDE --delete $WATCH_DIR/ $SSH_USER@$SSH_HOST:$REMOTE_TARGET_DIR"
+if [ `uname` == "Linux" ];then
+    WATCHCOMMAND=$WATCHCOMMAND_LINUX
+elif [ `uname` == "Darwin" ];then
+    WATCHCOMMAND=$WATCHCOMMAND_OSX
+else
+    echo "This will work only on Linux and OSX"
```
Comment (fago) [inline syncd]: So no #c clone?  would be great to clarify windows support in the readme

File: syncd.conf [hunk 6]
```diff
@@ -1,26 +1,25 @@
...
-
-# Sync options
-REMOTE_TARGET_DIR="/var/www"
-RSYNC_EXCLUDE='sites/default/files --exclude files'
-RSYNC_OPTIONS="-Cra --out-format='[%t]--%n' --include core"
-
-# Syncd files
+#######################################################################################
+#																					  #
+#		SyncDaemon, 2015, DataSyntax PE, ross@datasyntax.ua						 	  #
```
Comment (fago) [inline syncd.conf]: let's keep copyright stuff out of the config, it should be enough else, not?

**Comment by fago** [comment]
wow, thanks for this great PR!

I added few remarks (see above), would be great if we could get this resolved and merged!

File: syncd [hunk 7]
```diff
@@ -70,7 +89,7 @@ case $1 in
     ;;
 
     run)
-      $COMMAND && echo Done.
+        eval $SYNCCOMMAND && echo Done.
```
Comment (fago) [inline syncd]: Why is having eval good here?

File: syncd [hunk 8]
```diff
@@ -33,10 +52,10 @@ case $1 in
...
         exit 1;
       fi
-      export WATCH_EXCLUDE
-      export WATCH_VERBOSE
-      export WATCH_DIR
-      $SCRIPT_DIR/watch.sh $COMMAND >> $LOGFILE &
+      export WATCHCOMMAND
+      export VERBOSE
+      export LOGFILE
+      $SCRIPT_DIR/watch.sh $SYNCCOMMAND &
```
Comment (fago) [inline syncd]: so log support is gone?

**Pull request closed without merging**